filecoin-proofs/src/param.rs

```diff
@@ -118,117 +52,90 @@ pub fn get_parameter_digest(parameter_id: &str) -> Result<String> {
 
-        Ok(cid.as_str().to_string())
-    }
-}
+        let _ = stdout().flush();
+        let mut s = String::new();
+        stdin().read_line(&mut s).expect(ERROR_STRING);
 
-pub fn spawn_fetch_parameter_file(
-    is_verbose: bool,
-    parameter_map: &ParameterMap,
-    parameter_id: &str,
-    gateway: &str,
-) -> Result<()> {
-    let parameter_data = get_parameter_data(parameter_map, parameter_id)?;
-    let path = get_parameter_file_path(parameter_id);
-
-    create_dir_all(parameter_cache_dir())?;
-
-    let mut paramfile = match File::create(&path).map_err(failure::Error::from) {
-        Err(why) => return Err(why),
-        Ok(file) => file,
-    };
-
-    let client = Client::new();
-    let url = Url::parse(&format!("{}/ipfs/{}", gateway, parameter_data.cid))?;
-    let total_size = {
-        let res = client.head(url.as_str()).send()?;
-        if res.status().is_success() {
-            res.headers()
-                .get(header::CONTENT_LENGTH)
-                .and_then(|ct_len| ct_len.to_str().ok())
-                .and_then(|ct_len| ct_len.parse().ok())
-                .unwrap_or(0)
-        } else {
-            return Err(failure::err_msg("failed to download parameter file"));
+        match s.trim().to_uppercase().as_str() {
+            "Y" => return true,
+            "N" => return false,
+            _ => {}
         }
-    };
-
-    let req = client.get(url.as_str());
-    if is_verbose {
-        let mut pb = ProgressBar::new(total_size);
-        pb.set_units(Units::Bytes);
-
-        let mut source = FetchProgress {
-            inner: req.send()?,
-            progress_bar: pb,
-        };
-
-        let _ = copy(&mut source, &mut paramfile)?;
-    } else {
-        let mut source = req.send()?;
-        let _ = copy(&mut source, &mut paramfile)?;
     }
+}
 
-    Ok(())
+// Predicate which matches the provided extension against the given filename
+pub fn has_extension<S: AsRef<str>, P: AsRef<Path>>(filename: P, ext: S) -> bool {
+    filename
+        .as_ref()
+        .extension()
+        .and_then(OsStr::to_str)
+        .map(|s| s == ext.as_ref())
+        .unwrap_or(false)
 }
 
-pub fn validate_parameter_file(parameter_map: &ParameterMap, parameter_id: &str) -> Result<bool> {
-    let parameter_data = get_parameter_data(parameter_map, parameter_id)?;
-    let digest = get_parameter_digest(parameter_id)?;
+/// Builds a map from filename (in cache) to metadata.
+pub fn parameter_id_to_metadata_map<S: AsRef<str>>(
+    filenames: &[S],
+) -> Result<BTreeMap<String, CacheEntryMetadata>> {
+    let mut map: BTreeMap<String, CacheEntryMetadata> = Default::default();
 
-    if parameter_data.digest != digest {
-        Ok(false)
-    } else {
-        Ok(true)
-    }
-}
+    for filename in filenames {
+        let is_meta = PathBuf::from(filename.as_ref())
+            .extension()
+            .and_then(OsStr::to_str)
+            .map(|s| s == PARAMETER_METADATA_EXT)
+            .unwrap_or(false);
```

Comment:
why not use the `has_extension` from above

Reply:
Good call. Updated.